### PR TITLE
sql_test_connection for Big Query

### PR DESCRIPTION
**Pull request by fakirAyoub**
- `sql_test_connection` method for validating Authorisation for Big Query
- Tests
- Small Code Refactoring

### CODE REVIEW

File: packages/bigquery/sodasql/dialects/bigquery_dialect.py [hunk 1]
```diff
@@ -61,6 +62,33 @@ def create_connection(self):
...
+            LIMIT 1
+        """
+        return query
+
+    def sql_test_connection(self, dataset_id) -> Union[Exception, bool]:
+        logging.info(f'Listing tables to check connection')
+        try:
+            tables = self.client.list_tables(dataset_id)
+            if tables:
+                print("Tables contained in '{}':".format(dataset_id))
```
Comment (vijaykiran) [inline packages/bigquery/sodasql/dialects/bigquery_dialect.py]: Please use logging, instead  of `print` (also in other places in the file)

**Review by vijaykiran — CHANGES_REQUESTED** [review]

File: packages/bigquery/sodasql/dialects/bigquery_dialect.py [hunk 2]
```diff
@@ -61,6 +62,33 @@ def create_connection(self):
...
+            LIMIT 1
+        """
+        return query
+
+    def sql_test_connection(self, dataset_id) -> Union[Exception, bool]:
+        logging.info(f'Listing tables to check connection')
+        try:
+            tables = self.client.list_tables(dataset_id)
+            if tables:
+                logging.info("Tables contained in '{}':".format(dataset_id))
```
Comment (vijaykiran) [inline packages/bigquery/sodasql/dialects/bigquery_dialect.py:78]: We use versions of python that has format strings, you can change this to:

`f'Listing tables in {dataset_id}'`, same for other `"{}".format(...)` stuff below.

**Review by vijaykiran — CHANGES_REQUESTED** [review]

File: packages/bigquery/sodasql/dialects/bigquery_dialect.py [hunk 3]
```diff
@@ -61,6 +62,33 @@ def create_connection(self):
...
+                for table in tables:
+                    logging.info("{}.{}.{}".format(table.project, table.dataset_id, table.table_id))
+                    try:
+                        self.client.query(self.__query_table(table))
+                        return True
+                    except Exception as e:
+                        raise Exception("Unable to query table: {} from the dataset: {}. Exception: {}".format(table, dataset_id, e))
+                return True
+            else:
+                raise Exception("{} dataset does not contain any tables".format(dataset_id))
```
[the remark this answers is not in the document]
Comment (fakirAyoub) [reply packages/bigquery/sodasql/dialects/bigquery_dialect.py:88]: Thanks. This was solved in another commit after the PR creation

File: packages/bigquery/sodasql/dialects/bigquery_dialect.py (hunk 1 above)
Comment (fakirAyoub) [reply packages/bigquery/sodasql/dialects/bigquery_dialect.py]: Thanks. This was solved in another commit after the PR creation

**Pull request merged**